src/java.base/share/native/libjli/java.c

```diff
@@ -326,6 +329,41 @@ JLI_Launch(int argc, char ** argv,              /* main argc, argv */
         }
     }
 
+    {
+        /* Process -Xmso in the OPENJ9_JAVA_OPTIONS environment variable to
+         * set the main thread stack size. May be overridden by a later command
+         * line option.
+         */
+        char *str = getenv("OPENJ9_JAVA_OPTIONS");
+        if (str != NULL) {
+            while (*str != '\0') {
+                if ((*str == '-') && (JLI_StrCCmp(str + 1, "Xmso") == 0)) {
+                    char *msoSize = NULL;
+                    char *end = str + 5;
+                    str += 5;
+                    while ((*end != '\0') && !isspace(*end)) {
+                        end++;
+                    }
+                    msoSize = JLI_MemAlloc(end - str + 1);
+                    if (msoSize != NULL) {
+                        jlong tmp = 0;
+                        memcpy(msoSize, str, end - str);
+                        msoSize[end - str] = '\0';
+                        if (parse_size(msoSize, &tmp)) {
+                            threadStackSize = tmp;
+                            if (threadStackSize > 0 && threadStackSize < (jlong)STACK_SIZE_MINIMUM) {
+                                threadStackSize = STACK_SIZE_MINIMUM;
+                            }
+                        }
+                        JLI_MemFree(msoSize);
+                    }
+                    continue; /* str already incremented */
+                }
+                str++;
```

Comment:
I think this needs to be a little more sophisticated (with some of the features of `parseOptionsBuffer()`) to avoid mishandling quotes, etc. and values like this: `-Xmso64k -Dchild.stack=-Xmso32k` (the second occurrence of `-Xmso` should not apply).

Reply:
Updated.

Reply:
As discussed offline, this is better, but doesn't handle quotes (and escapes) like the VM. Quotes need not be just at the beginning and end of elements of the value (e.g. `-Xmso64k -Dchild.options="-Xint -Xmso32k -Xms4M"` would not be processed properly).

Reply:
I've updated it to re-use the parsing code in libjli/args.c